beacon_node/lighthouse_network/src/rpc/methods.rs

```diff
@@ -574,7 +574,7 @@ impl<T: EthSpec> std::fmt::Display for RPCResponse<T> {
                 write!(
                     f,
                     "LightClientBootstrap Slot: {}",
-                    bootstrap.header.beacon.slot
+                    bootstrap.header.beacon().slot
```

Comment:
With the update to superstruct, `beacon` can only be accessed via method call (unless we convert `header` to a specific `LightClientHeader` variant type)

This causes linting issues further downstream 
e.g. 

`large size difference between variants`

```
pub enum RPCCodedResponse<T: EthSpec> {
425 | |     /// The response is a successful.
426 | |     Success(RPCResponse<T>),
    | |     ----------------------- the largest variant contains at least 552 bytes
427 | |
428 | |     Error(RPCResponseErrorCode, ErrorType),
    | |     -------------------------------------- the second-largest variant contains at least 25 bytes
...   |
431 | |     StreamTermination(ResponseTermination),
432 | | }
    | |_^ the entire enum is at least 552 bytes
    
    help: consider boxing the large fields to reduce the total size of the enum
    |
426 |     Success(Box<RPCResponse<T>>),
```

If I follow clippys suggestions and Box `RPCResponse` this forces me to make some larger scale changes downstream to make the compiler happy. I'm a bit hesitant to make changes here, maybe we should ignore clippys suggestions here?

Reply:
I think it's due to `RPCResponse::LightClientBootstrap` variant here:
https://github.com/sigp/lighthouse/blob/03063bef268ce56d85658313030d511caa49bc49/beacon_node/lighthouse_network/src/rpc/methods.rs#L393

An enum size is the largest value of all variants, so we'd usually avoid having large objects directly stored in enum.

I think can just `Arc` the `LightClientBootstrap<T>` object here, like how it's done for blocks and blobs, since we only need a reference later to construct the SSZ object here:

https://github.com/sigp/lighthouse/blob/8a599ec7dc978f8d2a49100e3d712794b56eba2e/beacon_node/lighthouse_network/src/rpc/codec/ssz_snappy.rs#L76

Reply:
nice, that fixed it, thanks!